[Brandon Quek] iP

qkbrandon opened the pull request.

Commit:
Level-0

Commit:
Add command echoing feature

Commit:
Add ability to store and display text by user

Commit:
Add ability to mark tasks as done

Commit:
Edit to follow coding standard

Commit:
Add ToDos, Deadlines and Events features

Commit:
Edit runtest.bat

To accomodate symbols used in Task.java

Commit:
Add input.txt and EXPECTED.txt

Commit:
Edit to follow coding standard

JiawenLyu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+        System.out.println(greet);
+        
+        Task currentTask;
+        Task[] storeTasks = new Task[100];
```
Should the size 100 be extracted as a constant?

qkbrandon replied:
Thank you! will edit it later

JiawenLyu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+            }
+
+            if (line.contains("deadline")) {
+                String[] word = line.split("/by");
```
Perhaps use the name "words" rather than "word"?

qkbrandon replied:
Thank you! will edit it later

JiawenLyu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+            if (line.contains("done")) {
+                String[] word = line.split(" ");
+                int taskNum = Integer.parseInt(word[1]);
+                if (taskNum >= 1 || taskNum <= numOfTasks) {
```
I like the way of judging the conditions of "done" command!

qkbrandon replied:
Thank you!

tammykoh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String greet = showWelcomeScreen();
+        System.out.println(greet);
```
You could print greet in the showWelcomeScreen() function

qkbrandon replied:
Thank you! will edit it later

tammykoh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+
 public class Duke {
+
     public static void main(String[] args) {
```
Instead of having a chunk of code in main, you could put the code in functions and call the functions from main. Have a look at week 3 lecture on single level of abstraction per function (SLAP).

qkbrandon replied:
Thank you! will edit it later

tammykoh reviewed (commented):
Well done. I've included certain suggestions for you that could improve you code.

Commit:
Refactor code

Duke class: main function contains codes for features (e.g. todo)

By extracting the features' code into methods, there will be less
lines of code in main function and readability increases

Let's apply SLAP to reduce line of codes in main function

Also introduced constant for maximum number of tasks and
deleted redundant codes

Commit:
Add exception handling

Exception handling for empty todo
Exception handling for words not in command list

Commit:
Merge branch 'branch-Level-5'

Commit:
Put all classes in duke package

Commit:
Merge branch 'branch-A-Packages'

Commit:
Edit text-ui-test folder to accomodate 'duke' package

Commit:
Add delete feature

Commit:
Update input.txt and EXPECTED.txt to account for delete test cases

Commit:
Add duke.txt

Store tasks in text file

Commit:
Add feature to save tasks in txt automatically when tasks list change

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java

Retained both changes in branch-Level-6 and branch-Level-7

Commit:
Fix .JAR not recognising file at start up

yueyeah commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,343 @@
+        while (true) {
+            listCommand(storeTasks);
+            line = in.nextLine();
+            if (line.equals("bye")) {
```
"bye", "list", "done"....maybe you can define them as constants / enums?

yueyeah commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,343 @@
+
+            String[] words = line.split("todo ");
+            storeTasks.add(new ToDo(words[1]));
+            //storeTasks[numOfTasks] = new ToDo(words[1]);
```
This seems like past code that is commented out when not needed....you may want to delete this. Do make full use of version control to keep track of past code.

yueyeah commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,343 @@
+    }
+
+    private static void showWelcomeScreen() {
+        String greet = " ____________________________________________________________\n"
```
Can be better named as "greeting". Also, it can be defined as 1 huge constant as well.

yueyeah commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,343 @@
+    public static void main(String[] args) {
+        showWelcomeScreen();
+
+        ArrayList<Task> storeTasks = new ArrayList<>();
```
Maybe consider renaming storeTasks to taskList, task-<Noun> etc., like the format of tasksText in line 17.

yueyeah reviewed (commented):
You may want to consider converting your magic numbers/strings into constants. Also Duke.java is huge - in your future projects, you may want to consider a more OOP-style of using smaller class files instead of putting all code into a single monolithic class. Kaiwen's and Yu Hang's code is a great example of OOP-style - multiple small parts coming together to form a modular structure that can be modified easily.

Commit:
Add more error handling for various features

Features include: todo, event, deadline, done & delete

Commit:
Update input.txt & expected.txt for testing

Commit:
Refactor code

Add Ui, Storage, Parser & TaskList classes

Commit:
Add recognition for dates and times in tasks

Commit:
Update input.txt & expected.txt for testing

Commit:
Add search feature for tasks

Commit:
Update input.txt & expected.txt for testing

Commit:
Add JavaDoc comments to the code

qkbrandon committed:
Merge pull request #1 from bqxy/branch-Level-8

Merge branch-Level-8

qkbrandon committed:
Merge pull request #2 from bqxy/branch-Level-9

Merge branch-Level-9

qkbrandon committed:
Merge branch 'master' into branch-A-JavaDoc

qkbrandon committed:
Merge pull request #3 from bqxy/branch-A-JavaDoc

Merge branch-A-JavaDoc

Commit:
Add JavaDoc comments to the code

Commit:
Add User Guide (README.md)

Commit:
Revert "Add User Guide (README.md)"

This reverts commit c86a8417dfc04fda6ed9fb8b5a474d1755731db0.

Commit:
Add User Guide (README.md)
